cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -148,16 +157,39 @@ public void setProperties(final MetadataScope scope, final MetadataEntity metada
                  new MetadataRecordV2(metadataEntity, scope, propDeletions.build(), EMPTY_TAGS));
   }
 
+  private Boolean hasEntityInV1(MetadataScope scope, MetadataEntity metadataEntity) {
+    return execute(mds -> !mds.getProperties(metadataEntity).isEmpty() ||
+      !mds.getTags(metadataEntity).isEmpty(), scope, DATASET.V1);
+  }
+
+  private void updateInMemoryState() {
+    // update in memory variable hasV1SystemDs, hasV1BusinessDs
+    try {
```

Comment:
Can we try to throttle this to check only once every 5 seconds or so? Otherwise we may flood the DS service with requests.

Reply:
Hmm good point, but all the writes to metadata table would require a check for an existing entity. If we try to access a non existing table because of stale values of `hasV1SystemDs` or `hasV1BusinessDs`, the write to v2 table would also not succeed right?